src/controllers/__tests__/authenticationController.test.ts

```diff
@@ -206,13 +195,32 @@ const getClientProfileRedirectionUrl = (token: string): UrlFromString => {
   };
 };
 
-const controller = new AuthenticationController(
-  redisSessionStorage,
-  samlCert,
-  spidStrategyInstance,
-  tokenService,
-  getClientProfileRedirectionUrl
-);
+let controller: AuthenticationController;
+beforeAll(async () => {
+  if (!process.env.IDP_METADATA_URL) {
+    // TODO: Env variable will not load on local test env
+    process.env.IDP_METADATA_URL =
+      "https://registry.spid.gov.it/metadata/idp/spid-entities-idps.xml";
```

Comment:
ideally our tests shouldn't depend on external services - you could add a test xml to this repo and access it with the raw github URL

Reply:
even better would be to serve the file locally during the tests

Reply:
fixed in commit ebb3cf5